Build breaking because of type of function globDir

In package Glob there was a change in version 0.9.0, from 2017-10-01:

> Changed function types:
> System.FilePath.Glob.globDir :: [Pattern] -> FilePath -> IO [[FilePath]]
> No longer returns unmatched paths, like globDir1.

This seems to be breaking the build with following error:

```
hs-java-0.3.4/Java/ClassPath.hs:24:3: error:
        • Couldn't match expected type ‘[[FilePath]]’
                      with actual type ‘(t1 [FilePath], t0)’
        • In the pattern: (matches, _)
          In a stmt of a 'do' block:
            (matches, _) <- globDir (map compile patterns) dir
          In the expression:
            do { (matches, _) <- globDir (map compile patterns) dir;
                 return $ concat matches }
```

 Versions of Glob prior to 0.9.0 seem to be working okay.

--- a/JVM/Builder/Monad.hs
+++ b/JVM/Builder/Monad.hs
@@ -68,7 +68,7 @@ class Monad m => GeneratorMonad m where
   getGState :: m GState
   putGState :: GState -> m ()
 
-instance MonadState GState m => GeneratorMonad m where
+instance (Monad m, MonadState GState m) => GeneratorMonad m where
   getGState = St.get
   putGState = St.put
 

--- a/JVM/Builder/Monad.hs
+++ b/JVM/Builder/Monad.hs
@@ -88,7 +88,7 @@ class (Monad (g e), GeneratorMonad (g e)) => Generator e g where
 -- | Generate monad
 newtype Generate e a = Generate {
   runGenerate :: EMT e (State GState) a }
-  deriving (Monad, MonadState GState)
+  deriving (Functor, Applicative, Monad, MonadState GState)
 
 -- instance GeneratorMonad (Generate e) where
 --   getGState = Generate $ St.get
@@ -105,7 +105,7 @@ instance MonadState st (EMT e (State st)) where
 -- | IO version of Generate monad
 newtype GenerateIO e a = GenerateIO {
   runGenerateIO :: EMT e (StateT GState IO) a }
-  deriving (Monad, MonadIO, MonadState GState)
+  deriving (Functor, Applicative, Monad, MonadIO, MonadState GState)
 
 -- instance GeneratorMonad (GenerateIO e) where
 --   getGState = GenerateIO $ St.get
@@ -153,7 +153,7 @@ addItem c = do
           i' = if long c
                  then i+2
                  else i+1
-      modifyGState $ \st -> 
+      modifyGState $ \st ->
             st {currentPool = pool',
                 nextPoolIndex = i'}
       return i
@@ -259,7 +259,7 @@ endMethod = do
     Just method -> do
       let method' = method {methodAttributes = AR $ M.fromList [("Code", encodeMethod code)],
                             methodAttributesCount = 1}
-      modifyGState $ \st -> 
+      modifyGState $ \st ->
                st {generated = [],
                    currentMethod = Nothing,
                    doneMethods = doneMethods st ++ [method']}
@@ -377,4 +377,3 @@ generate cp name gen =
         superClass = "java/lang/Object",
         classMethodsCount = fromIntegral $ length (doneMethods res),
         classMethods = doneMethods res }
-

--- a/Java/ClassPath.hs
+++ b/Java/ClassPath.hs
@@ -21,7 +21,7 @@ import Java.JAR.Archive
 -- | For given list of glob masks, return list of matching files
 glob :: FilePath -> [FilePath] -> IO [FilePath]
 glob dir patterns = do
-  (matches, _) <- globDir (map compile patterns) dir
+  matches <- globDir (map compile patterns) dir
   return $ concat matches
 
 -- | Append one file to ClassPath forest
@@ -97,4 +97,3 @@ getEntry cp path = get cp (split "/" path)
       | toString (thisClass c) == path = return (Just i)
       | otherwise = get es [p]
     get x y = fail $ "Unexpected arguments for ClassPath.getEntry.get: " ++ show x ++ ", " ++ show y
-

